client/landing/stepper/declarative-flow/internals/steps-repository/free-post-setup/index.tsx

```diff
@@ -31,13 +31,16 @@ const FreePostSetup: Step = ( { navigation } ) => {
 	const [ selectedFile, setSelectedFile ] = useState< File | undefined >();
 	const [ isLoading, setIsLoading ] = useState( false );
 	const [ isSubmitError, setIsSubmitError ] = useState( false );
-
+	const [ siteDataFetched, setSiteDataFetched ] = useState( false );
 	const { saveSiteSettings } = useDispatch( SITE_STORE );
 
 	useEffect( () => {
-		setComponentSiteTitle( site?.name || '' );
-		setTagline( site?.description || '' );
-	}, [ site ] );
+		if ( site && ! siteDataFetched ) {
```

Comment:
Is it worthwhile to break this out into a reusable hook?

Reply:
I think that's a good idea